crux-calcite/src/crux/calcite.clj

```diff
@@ -0,0 +1,180 @@
+
+(extend-protocol OperandToCruxInput
+  RexInputRef
+  (operand->v [this schema]
+    (get-in schema [:crux.sql.table/query :find (.getIndex this)]))
+
+  org.apache.calcite.rex.RexCall
+  (operand->v [this schema]
+    (case (str (.-op this))
+      "CRUXID"
+      (keyword (operand->v (first (.-operands this)) schema))
+      (operand->v (first (.-operands this)) schema)))
+
+  RexLiteral
+  (operand->v [this schema]
+    (.getValue2 this)))
+
+(defn- ->operands [schema ^RexCall filter*]
+  (->> (.getOperands filter*)
+       (map #(operand->v % schema))))
+
+(defn- ->crux-where-clauses
+  [schema ^RexNode filter*]
+  (condp = (.getKind filter*)
+    SqlKind/EQUALS
+    [[(apply list '= (->operands schema filter*))]]
+    SqlKind/NOT_EQUALS
+    [(list 'not [(apply list '= (->operands schema filter*))])]
+    SqlKind/AND
+    (mapcat (partial ->crux-where-clauses schema) (.-operands ^RexCall filter*))
+    SqlKind/OR
+    [(apply list 'or (mapcat (partial ->crux-where-clauses schema) (.-operands ^RexCall filter*)))]
+    SqlKind/INPUT_REF
+    [[(list '= (operand->v filter* schema) true)]]
+    SqlKind/NOT
+    [(apply list 'not (mapcat (partial ->crux-where-clauses schema) (.-operands ^RexCall filter*)))]
+    SqlKind/GREATER_THAN
+    [[(list '> (->operands schema filter*))]]
+    SqlKind/GREATER_THAN_OR_EQUAL
+    [[(apply list '>= (->operands schema filter*))]]
+    SqlKind/LESS_THAN
+    [[(apply list '< (->operands schema filter*))]]
+    SqlKind/LESS_THAN_OR_EQUAL
+    [[(apply list '<= (->operands schema filter*))]]
+    SqlKind/LIKE
+    [[(apply list 'like (->operands schema filter*))]]
+    SqlKind/IS_NULL
+    [[(apply list '= (->operands schema filter*))]]
+    SqlKind/IS_NOT_NULL
+    [[(list 'not (apply list '= (->operands schema filter*)))]]))
+
+(defn- ->crux-query
+  [schema filters projects]
+  (try
+    (let [{:keys [crux.sql.table/columns crux.sql.table/query]} schema
+          {:keys [find where]} query]
+      {:find (if (seq projects) (mapv find projects) find)
+       :where (vec
+               (concat
+                (mapcat (partial ->crux-where-clauses schema) filters)
+                where))
+       :args [{:like #(org.apache.calcite.runtime.SqlFunctions/like %1 %2)}]})
+    (catch Throwable e
+      (log/error e)
+      (throw e))))
+
+(def ^:private column-types->sql-types {:varchar SqlTypeName/VARCHAR
+                                        :keyword SqlTypeName/VARCHAR
+                                        :integer SqlTypeName/INTEGER
+                                        :long SqlTypeName/BIGINT
+                                        :boolean SqlTypeName/BOOLEAN
+                                        :double SqlTypeName/DOUBLE
+                                        :datetime SqlTypeName/DATE})
+
+(defn- ^java.util.List perform-query [node q]
+  (->> (crux/q (crux/db node) q)
+       (mapv to-array)))
+
+(defn- ^String ->column-name [c]
+  (string/replace (string/upper-case (str c)) #"^\?" ""))
```

Comment:
placeholder comment for 'do we need to do this?' discussion :)

Reply:
On the README if we're supporting this functionality

Reply:
Added to the readme